### DISCUSSION TITLE
Is there a way to stop a project/cron instead of delete a project?

### DISCUSSION THREAD

**Discussion by BKuasney** [PrefectHQ/prefect#5165] [General] [answered]
What I want to achieve is:
_When some tasks fail, I need the project_name to stop running._ 

I achieved similar behaviour by using `client.delete_project(project_name=project_name)`, but this command is too dangerous since I lose all the logs on the UI.

What I was wondering is, we have the button to turn off the schedule inside the UI, under /Flow, where we can just "deactivate" the schedule for a specific project. Can we do that by terminal or python? Because is exactly what I need, but programmatically.

**Comment by zanieb** [comment, marked as answer]
You can query for all unarchived flows in a project e.g.

```graphql
query {
  flow(where:{project: {name: {_eq: "example"}}, archived: {_eq: false}}) {
    id
  }
}
```

Then use those ids as the input to the `set_schedule_inactive` route to disable each schedule

**Reply by BKuasney** [reply]
perfect man! For future reference for others, here is a quick-code (ofc it can be optimised):
```
client = Client()

# get the id
query = '''query {flow(where:{project: {name: {_eq: "<project_name>"}}, archived: {_eq: false}}) {id}}'''
result = client.graphql(query)
project_id = '"' + result['data']['flow'][0]['id'] + '"'

# pause flow
query = "mutation {set_schedule_inactive(input: {flow_id: %s}) {success}}" % project_id
client.graphql(query)
```

**Reply by zanieb** [reply]
Note that you're actually retrieving flow ids, not project ids. And here you're just setting the schedule inactive for the first flow found. You'll want something like this (untested)

```python
client = Client()

# get the ids
query = '''query {flow(where:{project: {name: {_eq: "<project_name>"}}, archived: {_eq: false}}) {id}}'''
result = client.graphql(query)
flow_ids = {flow['id'] for flow in result['data']['flow']}

mutation = '''mutation {set_schedule_inactive(input: {flow_id: "%s"}) {success}}'''

# pause flows
for flow_id in flow_ids:
    client.graphql(mutation % flow_id)
````
